python_src/src/lamp_py/performance_manager/l1_rt_trips.py

```diff
@@ -0,0 +1,442 @@
+                            all_new_trips.c.static_stop_count
+                            != VehicleTrips.static_stop_count,
+                            all_new_trips.c.first_last_station_match
+                            != VehicleTrips.first_last_station_match,
+                        ),
+                        sa.literal(True),
+                    )
+                ],
+                else_=sa.literal(False),
+            ).label("to_update"),
+        )
+        .select_from(all_new_trips)
+        .join(
+            VehicleTrips,
+            sa.and_(
+                all_new_trips.c.trip_hash == VehicleTrips.trip_hash,
+                VehicleTrips.start_date == seed_start_date,
+            ),
+            isouter=True,
+        )
+    )
+
+    # perform update/insert opertions with dataframe pulled from RDS
+    update_insert_trips = db_manager.select_as_dataframe(update_insert_query)
+
+    if update_insert_trips.shape[0] == 0:
+        process_logger.add_metadata(
+            trip_insert_count=0,
+            trip_update_count=0,
+        )
+        process_logger.log_complete()
+        return
+
+    update_insert_trips["fk_static_timestamp"] = update_insert_trips[
+        "fk_static_timestamp"
+    ].astype("int64")
+    update_insert_trips["start_date"] = update_insert_trips[
+        "start_date"
+    ].astype("int64")
+    update_insert_trips["start_time"] = update_insert_trips[
+        "start_time"
+    ].astype("int64")
+    update_insert_trips["stop_count"] = update_insert_trips[
+        "stop_count"
+    ].astype("int64")
+    update_insert_trips["static_start_time"] = update_insert_trips[
+        "static_start_time"
+    ].astype("Int64")
+    update_insert_trips["static_stop_count"] = update_insert_trips[
+        "static_stop_count"
+    ].astype("Int64")
+    update_insert_trips = update_insert_trips.fillna(numpy.nan).replace(
+        [numpy.nan], [None]
+    )
+
+    update_mask = (
+        update_insert_trips["existing_trip_hash"].notna()
+    ) & numpy.equal(update_insert_trips["to_update"], True)
+
+    process_logger.add_metadata(
+        trip_update_count=update_mask.sum(),
+    )
+
+    if update_mask.sum() > 0:
+        update_cols = [
+            "trip_hash",
+            "trunk_route_id",
+            "stop_count",
+            "static_trip_id_guess",
+            "static_start_time",
+            "static_stop_count",
+            "first_last_station_match",
+        ]
+        db_manager.execute_with_data(
+            sa.update(VehicleTrips.__table__).where(
+                VehicleTrips.trip_hash == sa.bindparam("b_trip_hash"),
+            ),
+            update_insert_trips.loc[update_mask, update_cols].rename(
+                columns={"trip_hash": "b_trip_hash"}
+            ),
+        )
+
+    process_logger.log_complete()
+
+
+def process_trips(
+    db_manager: DatabaseManager, events: pandas.DataFrame
+) -> pandas.DataFrame:
```

Comment:
doesn't look like this function has a return statement.

Reply:
What the heck, pylint is asleep at the wheel...